src/SSHDebugPS/AD7/AD7PortSupplier.cs

```diff
@@ -44,18 +36,33 @@ public int GetPortSupplierName(out string name)
 
         public virtual int CanPersistPorts()
         {
+            return HR.S_FALSE;
+        }
+
+        int IDebugPortSupplierDescription2.GetDescription(enum_PORT_SUPPLIER_DESCRIPTION_FLAGS[] flags, out string text)
+        {
+            text = Description;
             return HR.S_OK;
         }
 
-        public int RemovePort(IDebugPort2 pPort)
+        public virtual int EnumPersistedPorts(BSTR_ARRAY portNames, out IEnumDebugPorts2 portEnum)
```

Comment:
>EnumPersistedPorts [](start = 27, length = 18)

Is there a reason we aren't implementing this for Docker? If we aren't going to, we should probably stop implementing IDebugPortSupplier3, and move all of that just to the SSH implementation.

Reply:
Actually, the SSH doesn't implement it either. So we should remove IDebugPortSupplier3 and its two methods from the base class, and either implement EnumPersistedPorts in the Docker side, or remove EnumPersistedPorts/CanPersistedPorts all together.

---
In reply to: [287536095](https://github.com/microsoft/MIEngine/pull/868#discussion_r287536095) [](ancestors = 287536095)

Reply:
With SSH returning true for CanPersistPorts, can i do this? I feel like i need to keep it in place. 


---
In reply to: [287537861](https://github.com/microsoft/MIEngine/pull/868#discussion_r287537861) [](ancestors = 287537861,287536095)

Reply:
The only way to have the SDM persist ports is to both implement IDebugPortSupplier3 and also return S_FALSE from CanPersistPorts. Otherwise the SDM will just call EnumPorts.

Reply:
In other words - there isn't really a point in implementing IDebugPortSupplier3 unless your implementation of CanPersistPorts return S_FALSE, at least in some cases.